app.js

```diff
@@ -314,19 +314,18 @@ const outputPortalReport = (req, res) => {
 
       try {
         const result = await client.query(query, endpointValues);
-        const additionalColumns = result.rows.map(row => row.key);
-        additionalColumns.sort().forEach(newCol => {
-          if (columns.indexOf(newCol) === -1) {
-            columns.push(newCol);
-          }
-        });
+        additionalColumns = additionalColumns.concat(result.rows.map(row => row.key).sort());
       } catch (error) {
         done();
         return res.error(err.message, 500);
       }
       console.timeEnd('explode');
     }
 
+    // [ ... new Set(<some_array>) ] is a way to make sure that all the values are unique and their order is preserved.
+    // .concat returns a new array, so we're not modifying baseColumns.
+    const columns = [ ...new Set(baseColumns.concat(additionalColumns)) ];
```

Comment:
Looks like `Set` isn't supported in Firefox yet.  (https://caniuse.com/#search=set) We just need to make sure people use Chrome or Safari.